gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/PublisherApi.java

```diff
@@ -0,0 +1,498 @@
+    return create(new ServiceApiSettings());
+  }
+
+  /**
+   * Constructs an instance of PublisherApi, using the given settings. The channels are created based
+   * on the settings passed in, or defaults for any settings that are not set.
+   */
+  public static PublisherApi create(ServiceApiSettings settings) throws IOException {
+    return new PublisherApi(settings);
+  }
+
+  private PublisherApi(ServiceApiSettings settings) throws IOException {
+    ServiceApiSettings internalSettings = ApiUtils.settingsWithChannels(settings,
+        SERVICE_ADDRESS, DEFAULT_SERVICE_PORT, ALL_SCOPES);
+    this.settings = internalSettings;
+    this.channel = internalSettings.getChannel();
+  }
+
+  // ==============================
+  // Resource Name Helper Functions
+  // ==============================
+
+  public static final String createProject(String project) {
+    return PROJECT_PATH_TEMPLATE.instantiate(
+        "project", project);
+  }
+
+  public static final String createTopic(String project, String topic) {
+    return TOPIC_PATH_TEMPLATE.instantiate(
+        "project", project,"topic", topic);
+  }
+
+
+  // ========
+  // Getters
+  // ========
+
+  public Channel getChannel() {
+    return channel;
+  }
+
+
+  // =============
+  // Service Calls
+  // =============
+
+  // ----- createTopic -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates the given topic with the given name.
+   *
+   * @param name The name of the topic. It must have the format
+   * `"projects/{project}/topics/{topic}"`. `{topic}` must start with a letter,
+   * and contain only letters (`[A-Za-z]`), numbers (`[0-9]`), dashes (`-`),
+   * underscores (`_`), periods (`.`), tildes (`~`), plus (`+`) or percent
+   * signs (`%`). It must be between 3 and 255 characters in length, and it
+   * must not start with `"goog"`.
+   */
+  public Topic createTopic(String name) {
+    Topic request =
+        Topic.newBuilder()
+        .setName(name)
+        .build();
+
+    return createTopic(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates the given topic with the given name.
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   */
+  public Topic createTopic(Topic request) {
+    return createTopicCallable().call(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates the given topic with the given name.
+   */
+  public ApiCallable<Topic, Topic> createTopicCallable() {
+    return ApiUtils.prepareIdempotentCallable(CREATE_TOPIC, settings).bind(channel);
+  }
+
+  // ----- publish -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Adds one or more messages to the topic. Returns NOT_FOUND if the topic does
```

Comment:
I understand the problem you are facing but by the end of the day if we plan to expose these generated classes directly to users  (and even more so, for simple services that we may have
them without hand crafted layer) then we have a problem, as, in my opinion, this
javadoc is not good enough (and I mentioned several reasons). 

If you are able to fix it in the proto layer, great! if not, maybe you want to consider a way to "override" the generated javadoc with hand-written one.